src/common/tagging/external/tag_group.go

```diff
@@ -0,0 +1,209 @@
+package external
+
+import (
+	"errors"
+	"fmt"
+	"io/ioutil"
+	"strconv"
+
+	"github.com/bridgecrewio/yor/src/common/logger"
+	"github.com/bridgecrewio/yor/src/common/structure"
+	"github.com/bridgecrewio/yor/src/common/tagging"
+	"github.com/bridgecrewio/yor/src/common/tagging/tags"
+	"github.com/bridgecrewio/yor/src/common/utils"
+	"gopkg.in/yaml.v2"
+)
+
+type TagGroup struct {
+	tagging.TagGroup
+	configFilePath  string
+	config          map[interface{}]interface{}
+	tagGroupsByName map[string][]Tag
+}
+
+type Tag struct {
+	tags.ITag
+	defaultValue string
+	filters      map[interface{}]interface{}
+	matches      []interface{}
+}
+
+func (t Tag) SatisfyFilters(block structure.IBlock, tagFilterDir string) bool {
+	newTags, existingTags := block.GetNewTags(), block.GetExistingTags()
+	newTags = append(newTags, existingTags...)
+	satisfyFilters := true
+	for filterKey, filterValue := range t.filters {
+		if filterKey == "tags" {
+			for filterTagKey, filterTagValue := range filterValue.(map[interface{}]interface{}) {
+				strFilterValue := filterTagValue
+				if val, ok := filterTagValue.(int); ok {
+					strFilterValue = strconv.Itoa(val)
+				}
+				foundFilterTag := false
+				for _, blockTag := range newTags {
+					if blockTag.GetKey() == filterTagKey && blockTag.GetValue() == strFilterValue {
+						foundFilterTag = true
+						break
+					}
+				}
+				satisfyFilters = satisfyFilters && foundFilterTag
+			}
+		}
+		if filterKey == "directory" {
+			if tagFilterDir != filterValue {
+				satisfyFilters = false
+				break
+			}
+		}
+	}
+	return satisfyFilters
+}
+
+func (t *TagGroup) InitExternalTagGroups(configFilePath string) {
+	t.configFilePath = configFilePath
+	t.tagGroupsByName = make(map[string][]Tag)
+	t.InitExternalTagGroup()
+
+}
+
+func (t *TagGroup) InitTagGroup(dir string, skippedTags []string) {
+	t.SkippedTags = skippedTags
+	t.Dir = dir
+}
+
+func (t *TagGroup) InitExternalTagGroup() {
+	configMap := make(map[interface{}]interface{})
```

Comment:
@tronxd Try to make this configMap a struct, with types. Would solve a lot of the unmarshalling you implemented yourself. Check out the example [here](https://github.com/go-yaml/yaml#example)